hand-written.md

```diff
@@ -1,29 +1,19 @@
-We are happy to announce the availability of Scala 2.12.0-RC2!
+We are happy to announce the availability of Scala 2.12.0!
 
-This RC fixes all reported regressions since 2.11.
-It will become the final by October 28th, unless we hear of any issues that block your upgrade to 2.12 before then!
+The Scala 2.12 compiler has been completely overhauled to make use of the new VM features available in Java 8!
+ - A trait is compiled to an interface with default methods for improved binary compatibility.
+ - We've unified the treatment of Scala's built-in function types and Single Abstract Method types (for overload resolution, type inference and code generation).
+ - Specifically, code generation for function literals uses `invokedynamic` to avoid emitting a classfile at compile time. Many of these features were developed in fruitful collaboration with the Dotty team.
 
+Lukas has delivered a powerful new optimizer with this release, built on earlier work by Miguel Garcia. Many more  (effectively) final methods, including those defined in objects and traits, are now inlined. As well, closure allocations, dead code and [box/unbox pairs](https://github.com/scala/scala/pull/4858) are eliminated more effectively.
```

Comment:
I actually think it would sound more professional to say "This release ships with a powerful new optimizer", and give credits (Miguel, me) at the end of the paragraph.